### ISSUE TITLE
Recursion: tiny inconsistency in one of the linked resources

### ISSUE THREAD

**Issue by Eorlanas** [TheOdinProject/curriculum#27402]
### Checks

- [X] This is not a duplicate of an existing issue (please have a look through our [open issues list](https://github.com/TheOdinProject/curriculum/issues) to make sure)
- [X] I have thoroughly read and understand [The Odin Project Contributing Guide](https://github.com/TheOdinProject/.github/blob/main/CONTRIBUTING.md)
- [ ] Would you like to work on this issue?

### Describe your suggestion

It's tiny, really: one of the linked resources in the linked lesson (this one [https://www.scaler.com/topics/fibonacci-series-in-javascript/](url)) has a small inconsistency in its examples where the pseudocode is wrong but the actual Javascript code is not (see picture) which may be worthwhile pointing out in the Odin page for it to avoid confusion.
![image](https://github.com/TheOdinProject/curriculum/assets/114264403/462b3958-65ff-4b17-850e-d956ee66423a)

### Path

Node / JS

### Lesson Url

https://www.theodinproject.com/lessons/javascript-recursion

**Comment by fabulousgk** [comment]
What do you see wrong with the code?  I put it in a codepen and seems to be correct to me...

https://codepen.io/FabulousGeek/pen/LYavVPJ?editors=0011

**Comment by Eorlanas** [comment]
> What do you see wrong with the code? I put it in a codepen and seems to be correct to me...
> 
> https://codepen.io/FabulousGeek/pen/LYavVPJ?editors=0011

The pseudo code has the two beginning if blocks as if n == 0 and if n == 1, while the actual code has if n == 1 and if n == 2

**Comment by fabulousgk** [comment]
Ahh, yeah I see the confusion.  It is weird, because throughout the article he starts with n=0 as the first spot.  I think this may be a math thing.  Anyway, the JS is correct, if anything the pseduo code is a bit confusiong.

**Comment by fabulousgk** [comment]
See here for more info, but yeah basically you have a math concept that is being translated then into usable JS. Here is where the difference is.  Per the math definition n=7 is 13, but the JS is not actually providing the math definition but the x number.  Anotherwords while by math the number 13 is n=7 it is actually the 8th number.  The article definitely muddles that distinction.  This is sort of like array indexing in a way.

https://en.wikipedia.org/wiki/Fibonacci_sequence

And look, he switches the notation from the start:
<img width="662" alt="Screen Shot 2024-02-19 at 7 58 10 PM" src="https://github.com/TheOdinProject/curriculum/assets/126961361/78bbb0ee-7d6c-4ff6-aa7a-f93e213eb2b3">


IMO, rather than a disclaimer, a different article needs to be sourced as this one is very loose with these terms.

Good eye!

**Comment by wise-king-sullyman** [comment]
Agree, if we can find a better article we should swap it out. Thanks @Eorlanas and @fabulousgk!

**Comment by fabulousgk** [comment]
@wise-king-sullyman I will take this issue and see what I can find.  If someone else knows a resource already they can have it.

**Comment by sean-garwood** [comment]
@fabulousgk I can grab this if you'd like. I'm on the [Using Git in the Real World lesson](https://www.theodinproject.com/lessons/ruby-using-git-in-the-real-world) and am looking for a quick and easy first PR.

[New source: w3docs.com](https://www.w3docs.com/learn-javascript/recursion-and-stack.html)
* Secondarily, we could point to this [CS50 short on recursion](https://www.youtube.com/watch?v=mz6tAJMVmfM) that is linked to on the Ruby path (which I'm on).

**Comment by fabulousgk** [comment]
While that is a good link, we need something that details the fibonachi recursion in particular since that is where the article is being used.  I also believe any linked article must avoid discussing memoization since that has not been introduced at this stage.

**Comment by sean-garwood** [comment]
Good point. Here's a [GeeksForGeeks article](https://www.geeksforgeeks.org/javascript-program-to-display-fibonacci-sequence-using-recursion/) that fits better.

**Comment by fabulousgk** [comment]
> Good point. Here's a [GeeksForGeeks article](https://www.geeksforgeeks.org/javascript-program-to-display-fibonacci-sequence-using-recursion/) that fits better. Also, I saw that the CS50 recursion video is linked to in [the previous lesson](https://www.theodinproject.com/lessons/javascript-recursive-methods#assignment).

Looks like a good replacement to me!.  Just need to wait on a maintainer to agree.  You can have the change when they do.

**Comment by MaoShizhong** [comment]
Thank you @fabulousgk for being willing to hand it over!
@sean-garwood assigning this to you. If you can open a PR then your proposed changes can be reviewed as necessary :rocket:
